### PR TITLE
Like feature

### PR DESCRIPTION
**Pull request by Angryman18**
> First thing, PLEASE READ THIS: [ReactPlay Code Review Checklist](https://github.com/reactplay/react-play/wiki/ReactPlay-Code-Review-Checklist)

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. 

# Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by atapas** [comment]
@nirmalkc please post a message here once the style part is done.

File: src/common/const/nhost.js [hunk 1]
```diff
@@ -1,5 +1,7 @@
-export const NHOST = {
-  AUTH_URL: `https://rgkjmwftqtbpayoyolwh.nhost.run/v1/auth/signin/provider/github?redirectTo=${encodeURI(
-    "http://localhost:3000/plays/create"
-  )}`,
+const AUTH_URL = (redirectURL, provider = "github") => {
+  return `https://rgkjmwftqtbpayoyolwh.nhost.run/v1/auth/signin/provider/${provider}?redirectTo=${encodeURI(
```
Comment (atapas) [inline src/common/const/nhost.js:2]: The URL should be taken from .env file.

The .env already has the fragment of this URL to use. Do not create any new env variable.

File: src/common/createplay/CreatePlay.jsx [hunk 2]
```diff
@@ -121,7 +121,7 @@ const CreatePlay = () => {
     if (refreshToken) {
       console.log(refreshToken);
     } else {
-      window.location = NHOST.AUTH_URL;
+      window.location = NHOST.AUTH_URL('http://localhost:3000/plays/create');
```
Comment (atapas) [inline src/common/createplay/CreatePlay.jsx:124]: Why?

Comment (atapas) [inline src/common/createplay/CreatePlay.jsx:124]: Also, it is not guaranteed that the dev will always run on the 3000 port. It should be from the environment variable and in code default to 3000

File: src/common/createplay/CreatePlay.jsx [hunk 3]
```diff
@@ -121,7 +121,7 @@ const CreatePlay = () => {
     if (refreshToken) {
       console.log(refreshToken);
```
Comment (atapas) [inline src/common/createplay/CreatePlay.jsx:122]: If the log statement is intentional, then make it more useful.

File: src/common/header/HeaderNav.jsx [hunk 4]
```diff
@@ -10,6 +10,7 @@ import { GoX } from "react-icons/go";
 import { Modal, Box, Typography, Menu } from "@mui/material";
 import { useContext } from "react";
 import { SearchContext } from "common/search/search-context";
+import LogOut from './Logout'
```
Comment (atapas) [inline src/common/header/HeaderNav.jsx:13]: Comment out

File: src/common/header/HeaderNav.jsx [hunk 5]
```diff
@@ -152,6 +153,7 @@ const HeaderNav = ({ showBrowse }) => {
               <span className="btn-label">Share</span>
             </button>
           </li>
+          // <LogOut />
```
Comment (atapas) [inline src/common/header/HeaderNav.jsx:156]: That's not how you comment in JSX.

**Review by atapas — CHANGES_REQUESTED** [review]
First set of the reviews

@Angryman18

File: src/common/header/HeaderNav.jsx (hunk 5 above)
Comment (Angryman18) [reply src/common/header/HeaderNav.jsx:156]: lol sorry.

File: src/common/const/nhost.js (hunk 1 above)
Comment (Angryman18) [reply src/common/const/nhost.js:2]: ah i was just tying with that and mistakenly pushed it also. fixing now.

Comment (Angryman18) [reply src/common/const/nhost.js:2]: ah this was hardcoded from previous. our current prod has also this hardcoded btw.

File: src/common/createplay/CreatePlay.jsx (hunk 2 above)
Comment (Angryman18) [reply src/common/createplay/CreatePlay.jsx:124]: this is bcz we need redirect user based on auth which is different in two different usecase. thats y we are passing redirect url dynamically. auth while liking a play (when user wants to like a play) we always want to redirect user to that specefic play after auth and slo for create play (after auth) we want to redirect user into /plays/create route.

Comment (Angryman18) [reply src/common/createplay/CreatePlay.jsx:124]: ok will fix this

File: src/common/modal/MuiModal.jsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  bgcolor: "background.paper",
+  borderRadius: 2,
+  boxShadow: 24,
+  border: 'none',
+  outline: 'none',
+  px: 4,
+  py: 2,
+};
+
+const MuiModal = ({ component, children, open, handleClose }) => {
```
Comment (koustov) [inline src/common/modal/MuiModal.jsx:18]: Let's not use any reserved/third-party keyword.

File: src/common/modal/MuiModal.jsx [hunk 7]
```diff
@@ -0,0 +1,30 @@
+import { Modal, Box } from "@mui/material";
```
Comment (koustov) [inline src/common/modal/MuiModal.jsx:1]: I don't think this component is being used.

File: src/common/playlists/PlayHeaderActions.jsx [hunk 8]
```diff
@@ -1,76 +1,159 @@
...
+import MuiModal from "common/modal/MuiModal";
+import useLikePlays from "common/hooks/useLikePlays";
+import { NHOST } from "common/const";
+
+// we cannot remove delete SignInMethods, MuiModal file since in future we will support
+// more authentication provider. so let us keep it.
+
+
+// we are not getting latest play_like object after liking or updating liking a play
+// thats y we need to track interations of the user
```
Comment (koustov) [inline src/common/playlists/PlayHeaderActions.jsx:20]: `y` => `why`

File: src/common/playlists/PlayHeaderActions.jsx [hunk 9]
```diff
@@ -1,76 +1,159 @@
...
 
 const PlayHeaderActions = ({ play }) => {
-  console.log(play);
+  const { play_like } = play;
+  const userId = useUserId();
+
+  const { likePlay, unLikePlay } = useLikePlays();
+
   const [showComment, setShowComment] = useState(false);
+  // const [showSignInMethods, setShowSignInMethods] = useState(false);
```
Comment (koustov) [inline src/common/playlists/PlayHeaderActions.jsx:34]: lets not have the commented code.

File: src/common/playlists/PlayHeaderActions.jsx [hunk 10]
```diff
@@ -1,76 +1,159 @@
...
+              <MdClose size={24} className='icon' />
             </button>
           </div>
-          <div className="comments-body">
+          <div className='comments-body'>
             <Comment />
           </div>
         </div>
       )}
+      {/* <MuiModal
```
Comment (koustov) [inline src/common/playlists/PlayHeaderActions.jsx:152]: Remove commented code

File: src/common/playlists/PlayThumbnail.jsx [hunk 11]
```diff
@@ -54,9 +67,16 @@ const PlayThumbnail = ({ play }) => {
     }
   }, [play]);
 
+  const countLikes = (Obj) => {
```
Comment (koustov) [inline src/common/playlists/PlayThumbnail.jsx:70]: MINOR: I see this method is repeated on multiple files. So we should create a util service and put this method inside. The method name could be `countByProp(obj, propName, propValue)` and call it as 

```javascript
countByProp(Obj, 'liked', true)
```

File: src/common/playlists/PlayThumbnail.jsx [hunk 12]
```diff
@@ -28,7 +30,18 @@ const Author = ({ user }) => {
 
 const PlayThumbnail = ({ play }) => {
   const [cover, setCover] = useState(null);
+  const isAuthenticated = useAuthenticated();
+  const userId = useUserId();
 
+  const LikeObject = () => {
```
Comment (koustov) [inline src/common/playlists/PlayThumbnail.jsx:36]: Need to be in camelCase

File: src/common/playlists/PlayHeaderActions.jsx [hunk 13]
```diff
@@ -1,76 +1,159 @@
...
 import { MdClose } from "react-icons/md";
-
+import Like from "common/components/Like/Like";
+import SignInMethods from "./SignInMethods";
 import Comment from "common/components/Comment";
+import MuiModal from "common/modal/MuiModal";
+import useLikePlays from "common/hooks/useLikePlays";
+import { NHOST } from "common/const";
+
+// we cannot remove delete SignInMethods, MuiModal file since in future we will support
```
Comment (koustov) [inline src/common/playlists/PlayHeaderActions.jsx:15]: We can add it back when we plan for it, and probably the mechanism would be different. So my suggestion to remove them.

File: src/common/playlists/SignInMethods.jsx [hunk 14]
```diff
@@ -0,0 +1,36 @@
+import { Typography, Box, Stack, Button } from "@mui/material";
```
Comment (koustov) [inline src/common/playlists/SignInMethods.jsx:1]: Even though, i saw the comment however i would still ask to remove it.

File: src/common/playlists/playlist.css [hunk 15]
```diff
@@ -462,6 +462,21 @@
 
 }
 
+.play-details .header-actions .action.counted {
+    display: flex;
+    width: auto;
+    height: auto;
+    text-decoration: none;
```
Comment (koustov) [inline src/common/playlists/playlist.css:469]: `none` is default value for text-decoration. isn't it? Or it's being used to overwrite some parent property?

**Review by koustov — CHANGES_REQUESTED** [review]
Few comments dropped

File: src/common/createplay/CreatePlay.jsx (hunk 2 above)
Comment (koustov) [reply src/common/createplay/CreatePlay.jsx]: I guess both should work with `window.location.href`. We can still maintain uniformity

File: src/common/playlists/PlayThumbnail.jsx [hunk 16]
```diff
@@ -4,6 +4,8 @@ import { BsPlayCircleFill } from "react-icons/bs";
 import thumbPlay from "images/thumb-play.png";
```
[the remark this answers is not in the document]
Comment (atapas) [reply src/common/playlists/PlayThumbnail.jsx:4]: I'M TALKING TO @nirmalkc ABOUT THIS ONE.

File: src/index.js [hunk 17]
```diff
@@ -32,11 +32,11 @@ const Index = () => {
     setShowShareModal,
   };
   return (
-    <React.StrictMode>
+    // <React.StrictMode>
```
Comment (atapas) [inline src/index.js:35]: This shouldn't be done.

File: src/common/services/request/query/like-play.js [hunk 18]
```diff
@@ -0,0 +1,45 @@
+export const LikeIndividualPlay = (Obj) => {
+  const { play_id, user_id } = Obj;
```
Comment (atapas) [inline src/common/services/request/query/like-play.js:2]: Argument shouldn't be cpital case

File: src/common/components/Like/Like.jsx [hunk 19]
```diff
@@ -0,0 +1,25 @@
...
+  const {liked, number} = likeObj
+  
+  const likeClickHandler = () => {
+    if (onLikeClick) return onLikeClick();
+  };
+  
+  return (
+    <button className="action counted -mr-0.5" onClick={likeClickHandler}>
+      <AiOutlineLike size='24px' className={`${liked ? "hidden" : "icon"}`}/>
+      <AiFillLike size='24px' className={`${liked ? "icon" : "hidden"}`}/>
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/common/components/Like/Like.jsx:14]: like how?

File: src/common/header/HeaderNav.jsx [hunk 20]
```diff
@@ -10,6 +10,7 @@ import { GoX } from "react-icons/go";
 import { Modal, Box, Typography, Menu } from "@mui/material";
 import { useContext } from "react";
 import { SearchContext } from "common/search/search-context";
+// import LogOut from './Logout'
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/common/header/HeaderNav.jsx:13]: removed

File: src/common/header/HeaderNav.jsx [hunk 21]
```diff
@@ -152,6 +153,7 @@ const HeaderNav = ({ showBrowse }) => {
               <span className="btn-label">Share</span>
             </button>
           </li>
+          {/* <LogOut /> */}
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/common/header/HeaderNav.jsx:156]: removed

File: src/common/header/HeaderNav.jsx [hunk 22]
```diff
@@ -10,6 +10,7 @@ import { GoX } from "react-icons/go";
 import { Modal, Box, Typography, Menu } from "@mui/material";
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/common/header/HeaderNav.jsx:10]: didnt understand.

File: src/common/header/Logout.jsx [hunk 23]
```diff
@@ -0,0 +1,24 @@
+import { IoLogOutSharp } from "react-icons/io5";
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/common/header/Logout.jsx:1]: removed

File: src/common/modal/MuiModal.jsx (hunk 6 above)
Comment (Angryman18) [reply src/common/modal/MuiModal.jsx:18]: removed

File: src/common/modal/MuiModal.jsx (hunk 7 above)
Comment (Angryman18) [reply src/common/modal/MuiModal.jsx:1]: removed

File: src/common/playlists/PlayHeaderActions.jsx (hunk 8 above)
Comment (Angryman18) [reply src/common/playlists/PlayHeaderActions.jsx:20]: removed

File: src/common/playlists/PlayHeaderActions.jsx (hunk 9 above)
Comment (Angryman18) [reply src/common/playlists/PlayHeaderActions.jsx:34]: removed

File: src/common/playlists/PlayHeaderActions.jsx (hunk 10 above)
Comment (Angryman18) [reply src/common/playlists/PlayHeaderActions.jsx:152]: removed

File: src/common/playlists/PlayHeaderActions.jsx (hunk 13 above)
Comment (Angryman18) [reply src/common/playlists/PlayHeaderActions.jsx:15]: removed

File: src/common/playlists/SignInMethods.jsx (hunk 14 above)
Comment (Angryman18) [reply src/common/playlists/SignInMethods.jsx:1]: removed

File: src/common/playlists/playlist.css (hunk 15 above)
Comment (Angryman18) [reply src/common/playlists/playlist.css:469]: i didnt make ui changes

File: src/index.js (hunk 17 above)
Comment (Angryman18) [reply src/index.js:35]: fixed

File: src/common/services/request/query/like-play.js (hunk 18 above)
Comment (Angryman18) [reply src/common/services/request/query/like-play.js:2]: fixed

File: src/common/services/request/query/like-play.js [hunk 24]
```diff
@@ -0,0 +1,45 @@
...
+    object: {
+      play_id: play_id,
+      user_id: user_id,
+      liked: true,
+    },
+    return: ["id"],
+  };
+};
+
+export const unlikeIndividualPlay = (Obj) => {
```
[the remark this answers is not in the document]
Comment (Angryman18) [reply src/common/services/request/query/like-play.js:17]: fixed

File: src/common/playlists/PlayThumbnail.jsx (hunk 16 above)
[the remark this answers is not in the document]
Comment (atapas) [reply src/common/playlists/PlayThumbnail.jsx:4]: Done

**Review by atapas — APPROVED** [review]

**Comment by atapas** [comment]
@koustov please do a final review.

I am approving it. You do a final review and merge if all is good. Please clean up the play_like table entries before you merge.. Also, you may have to clean up the nhost redirect URL entry for the preview URL.. finally do some sanity after you merge..

**Review by koustov — APPROVED** [review]
lgtm

**Pull request merged**